File: core/src/main/java/org/readutf/hermes/event/Listener.java
package org.readutf.hermes.event;

import org.readutf.hermes.platform.HermesChannel;
import org.readutf.hermes.packet.Packet;

public interface Listener<PACKET extends Packet<RESPONSE>, RESPONSE> {

    RESPONSE onPacket(HermesChannel hermesChannel, PACKET event) throws Exception;

}

File: core/src/main/kotlin/org/readutf/hermes/listeners/Listener.kt
package org.readutf.hermes.listeners

import com.github.michaelbull.result.Result
import org.readutf.hermes.Packet
import org.readutf.hermes.channel.HermesChannel

public interface Listener {
    public fun acceptPacket(
        hermesChannel: HermesChannel,
        packet: Packet<*>,
    ): Result<Any?, Throwable>
}
